frontend/.eslintrc

```diff
@@ -63,6 +63,14 @@
           {
             "group": ["~/api/**"],
             "message": "Read from '~/api' instead."
+          },
+          {
+            "group": ["~/components/table/**", "!~/components/table/useTableColumnSort"],
+            "message": "Read from '~/components/table' instead."
+          },
+          {
+            "group": ["~/components/table/useTableColumnSort"],
+            "message": "The data will be sorted in the table, don't use this hook outside of '~/components/table' repo. For more information, please check the props of the Table component."
```

Comment:
I think we should consider using a barrel file for our components API or adopt an `internal` folder structure hierarchy.
Not part of this PR, but for future tech debt. cc @andrewballantyne

Reply:
@andrewballantyne pointed me to this https://github.com/openshift/hac-infra#folder-structure-design which he suggests we adopt and I agree. I will make a separate tech debt item to capture this.

Reply:
https://github.com/opendatahub-io/odh-dashboard/issues/1833